File: ejercicio2/lambda.py
n = 10
fibo = lambda x : 0 if x == 0 else 0 if x == 1 else 1 if x == 2 else fibo(x - 1) + fibo(x - 2) + fibo(x - 3)
for i in range(1, n):
    print(f"{fibo(i)},", end="")
print(fibo(n))

File: ejercicio2/lambda.scala
object Fibonacci{
  def main(args:Array[String]):Unit={
  def fib(n:Int):BigInt = {
    def fibo(n:Int):(BigInt,BigInt) = if (n == 1) (1,0) else {
      val (a,b) = fibo(n/2)
      val p = (2*b+a)*a
      val q = a*a + b*b
      if(n % 2 == 0) (p,q) else (p+q,p)
    }
    fibo(n)._1
  }
  for (i <- 0 to 10){
      print(fib(i))
    }
  }
}